src/features/cleaner_file_type.rs

```diff
@@ -10,12 +10,16 @@ use super::utils::collect_metrics;
 
 pub fn directory_cleaner_based_on_file_type(
     dir: &String,
-    paths_to_clear: &[String],
+    types_to_clear: &[String],
     dry_run: bool,
     report_data: &mut ReportData,
+    paths_to_ignore: &[String],
 ) -> Result<()> {
     let mut del_count: u32 = 0;
     let mut del_size: u64 = 0;
+
+    // Convert paths_to_ignore to a collection of PathBuf for easier comparison
+    let ignore_set: Vec<PathBuf> = paths_to_ignore.iter().map(PathBuf::from).collect();
```

Comment:
I think collecting this into a set makes more sense instead of a vector. I'm not really keen on this as i don't see the client passing in so many args